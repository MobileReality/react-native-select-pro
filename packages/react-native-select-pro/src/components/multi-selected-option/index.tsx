export { MultiSelectedOption } from './multi-selected-option';
