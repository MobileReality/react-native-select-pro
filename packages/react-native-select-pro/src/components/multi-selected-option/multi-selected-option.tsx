import React, { memo } from 'react';
import isEqual from 'react-fast-compare';
import type { TextStyle, ViewStyle } from 'react-native';
import { Pressable, StyleSheet, Text } from 'react-native';

import { COLORS, FONT_SIZE, PRESSED_STYLE } from '../../constants';

import type { MultiSelectedOptionProps } from './multi-selected-option.types';

export const MultiSelectedOption = memo(
    <T,>({
        option,
        multiSelectedCustomStyles,
        optionWidth,
        onPressRemove,
        disabled,
    }: MultiSelectedOptionProps<T>) => {
        return (
            <Pressable
                accessibilityLabel={
                    option ? `${option.label} selected` : 'Placeholder in multiple select'
                }
                style={({ pressed }) => [
                    styles.multiSelectedOption,
                    multiSelectedCustomStyles?.container,
                    { width: optionWidth },
                    pressed && (multiSelectedCustomStyles?.pressed ?? PRESSED_STYLE),
                ]}
                disabled={disabled}
                onPress={() => (onPressRemove ? onPressRemove(option) : null)}
            >
                <Text
                    numberOfLines={1}
                    style={[
                        styles.text,
                        multiSelectedCustomStyles?.text,
                        {
                            color:
                                StyleSheet.flatten(multiSelectedCustomStyles?.text)?.color ??
                                COLORS.BLACK,
                        },
                    ]}
                >
                    {option?.label}
                </Text>
            </Pressable>
        );
    },
    isEqual,
);

type Styles = {
    text: TextStyle;
    multiSelectedOption: ViewStyle;
};

const styles = StyleSheet.create<Styles>({
    text: {
        fontSize: FONT_SIZE,
        textAlign: 'left',
    },
    multiSelectedOption: {
        alignItems: 'flex-start',
        justifyContent: 'center',
        display: 'flex',
        borderRadius: 4,
        backgroundColor: COLORS.DISABLED,
        borderWidth: 1,
        borderColor: COLORS.BLACK,
        margin: 2,
        paddingLeft: 5,
    },
});

MultiSelectedOption.displayName = 'MultiSelectedOption';
