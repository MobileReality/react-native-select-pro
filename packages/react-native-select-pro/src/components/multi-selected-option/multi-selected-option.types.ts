import type {
    OnPressRemove,
    OptionalToRequired,
    OptionType,
    SelectProps,
    SelectStyles,
} from '../../types';

export type MultiSelectedOptionProps<T> = {
    option: OptionType<T> | null;
    optionWidth: number | string;
    onPressRemove: OnPressRemove<T> | null;
    multiSelectedCustomStyles: NonNullable<SelectStyles['select']>['multiSelectedOption'];
} & OptionalToRequired<Pick<SelectProps<T>, 'disabled'>>;
