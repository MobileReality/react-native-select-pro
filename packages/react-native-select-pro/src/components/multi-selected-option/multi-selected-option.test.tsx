import React from 'react';
import { render } from '@testing-library/react-native';

import { MultiSelectedOption } from './multi-selected-option';

describe('MultiSelectedOption', () => {
    it('should generate MultiSelectedOption snapshot', () => {
        const multiSelectedOption = render(
            <MultiSelectedOption
                option={{ label: 'test', value: 'test' }}
                optionWidth="100%"
                isPlaceholder={false}
                placeholderText={undefined}
                placeholderTextColor={undefined}
                textStyle={{}}
                multiSelectionOptionStyle={{}}
            />,
        );
        expect(multiSelectedOption).toMatchSnapshot();
    });
});
