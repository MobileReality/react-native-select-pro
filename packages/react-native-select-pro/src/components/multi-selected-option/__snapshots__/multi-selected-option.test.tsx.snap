// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectedOption should generate MultiSelectedOption snapshot 1`] = `
<View
  accessibilityLabel="test selected"
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "flex-start",
        "backgroundColor": "#e0e0e0",
        "borderColor": "#000000",
        "borderRadius": 4,
        "borderWidth": 1,
        "display": "flex",
        "justifyContent": "center",
        "margin": 2,
        "paddingLeft": 5,
      },
      {},
      {
        "width": "100%",
      },
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "fontSize": 12,
          "textAlign": "left",
        },
        {},
        {
          "color": "#000000",
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
